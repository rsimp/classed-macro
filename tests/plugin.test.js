const pluginTester = require('babel-plugin-tester').default;
const plugin = require('babel-plugin-macros');

pluginTester({
  plugin,
  pluginName: "classed.macro",
  snapshot: true,
  babelOptions: {filename: __filename},
  tests: [
    `
    import tag from '../macro.js'

    const topFoo = tag.div\`foo foo\`;

    const topBar = tag.div\`bar bar\`;

    topFoo.defaultProps = {
      someProp: true
    };

    function makeFoo() {
      const foo = tag.div\`foo foo\`;
      return foo;
    }

    function makeBar() {
      const bar = tag.div\`bar bar\`;
      const otherStatement = 42;
      const defaultProps = {
        someProp: true
      };
      bar.defaultProps = defaultProps;
      return bar;
    }
    `,
  ],
});