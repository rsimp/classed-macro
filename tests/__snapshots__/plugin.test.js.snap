// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classed.macro 1. classed.macro: 1. classed.macro 1`] = `

import tag from '../macro.js'

const topFoo = tag.div\`foo foo\`;

const topBar = tag.div\`bar bar\`;

topFoo.defaultProps = {
  someProp: true
};

function makeFoo() {
  const foo = tag.div\`foo foo\`;
  return foo;
}

function makeBar() {
  const bar = tag.div\`bar bar\`;
  const otherStatement = 42;
  const defaultProps = {
    someProp: true
  };
  bar.defaultProps = defaultProps;
  return bar;
}

      ↓ ↓ ↓ ↓ ↓ ↓

import _classed from "classed-components";
const topFoo = _classed.div\`foo foo\`;
topFoo.displayName = "topFoo";
const topBar = _classed.div\`bar bar\`;
topBar.displayName = "topBar";
topBar.defaultProps = {};
topBar.defaultProps["data-react-component"] = "plugin.test__topBar";
topFoo.defaultProps = {
  someProp: true,
};
topFoo.defaultProps["data-react-component"] = "plugin.test__topFoo";

function makeFoo() {
  const foo = _classed.div\`foo foo\`;
  foo.displayName = "foo";
  foo.defaultProps = {};
  foo.defaultProps["data-react-component"] = "plugin.test__foo";
  return foo;
}

function makeBar() {
  const bar = _classed.div\`bar bar\`;
  bar.displayName = "bar";
  const otherStatement = 42;
  const defaultProps = {
    someProp: true,
  };
  bar.defaultProps = defaultProps;
  bar.defaultProps["data-react-component"] = "plugin.test__bar";
  return bar;
}


`;
